### PR TITLE
Receive 3vl

### PR DESCRIPTION
**Pull request by nealrichardson**
This PR tries to strike the right balance for our current stage in the logic variable work. The ostensible goal is "to receive 3VL (literals, vectors, and definitions), but convert them to boolean on receive and continue to both receive boolean and use boolean internally," though it is slightly more complicated than that. 

1. Definition of "logic variables": we have (I think?) decided that we will want to treat as logical variables and convert to R `logical` type any categorical that has any category with `"selected": true`, not just those with category names are "Selected", "Other", "No Data" or some other "three-valued" restriction. That implies a notable interface change in the R package in a few areas, including multiple response data (as in `as.vector`), which currently comes in as categorical. To limit the disruption (or rather, to add the API adaptation for the new backend and then separately tackle this feature in R), the function `is.3vl()` is more narrowly defined to include only the system type specification that the API will start returning as an alternative to the boolean type, and `as.vector()` on categorical_array and multiple_response is restricted to not return as R logical.

2. The pseudo-logical categorical variable currently created in R via `toVariable()` with categories "True" and "False". This PR introduces a slight change to the API for these variables, creating them as logical variables with ids 1, 0, and -1 and with a selected category but preserving the "True" and "False" names. Because their names differ from the system 3VL category names, the strict definition in `is.3vl()` will not determine that this categorical variable should be rendered as an R logical, keeping with the current behavior. We can introduce that change whenever, independent of the development of the backend.

In sum, this PR attempts to separate the behavior changes we want to introduce from the forward-compatible API client changes so that progress can be made on the server-side implementation independently. 

Still to do: add some test fixtures for 3VL data and confirm that it is consumed correctly. The only change that should be required to existing tests is that data coming down from logical-as-categorical variables should be 1, 0, -1 instead of 1, 2, -1. This is an internal detail that our users do not rely on, so it is safe to change here.

### CODE REVIEW

File: tests/testthat/test-update.R [hunk 1]
```diff
@@ -160,13 +160,9 @@ with_test_authentication({
...
         ds$v6[c(2, 5)] <- FALSE
-        expect_equal(as.vector(ds$v6[1:5], "id"), c(1, 2, 1, 1, 2))
+        expect_equal(as.vector(ds$v6[1:5], "id"), c(1, 0, 1, 1, 0))
         ds$v6[c(6, 9)] <- c(FALSE, NA)
         expect_equal(as.vector(ds$v6[1:10], "id"),
-            c(1, 2, 1, 1, 2, 2, 1, 1, -1, 1))
-    })
-    test_that("Can't edit a regular categorical with logical", {
-        expect_error(ds$v4[5] <- TRUE,
-            "Cannot update CategoricalVariable with type logical")
```
[the remark this answers is not in the document]
Comment (nealrichardson) [reply tests/testthat/test-update.R]: This is an integration test of validation that doesn't require making a request, and it's already tested in the mock tests above.

File: R/variable-to-R.R [hunk 2]
```diff
@@ -39,6 +44,10 @@ parse_column <- list(
         out <- values(cats)[match(out, ids(cats))]
         return(out)
     },
+    logical=function (col, variable, mode=NULL) {
+        out <- columnParser("numeric")(col)
+        return(as.logical(out))
```
[the remark this answers is not in the document]
Comment (nealrichardson) [reply R/variable-to-R.R]: Correct; `is.3vl()` is very strict right now, so you'll only be in here if you have `1, 0, -1`. I'll add a comment here though for whenever we decide to implement the logical behavior when `any(is.selected(categories(variable)))`.

File: R/cube-dims.R [hunk 3]
```diff
@@ -18,6 +18,13 @@ cubeDims <- function (cube) {
         }
         ## If enumerated, will be "elements", not "categories"
         d <- a$type$categories %||% a$type$elements
+        ## Sniff for 3VL
+        cats <- try(Categories(data=d), silent=TRUE)
+        if (is.categories(cats) && is.3vl(cats)) {
+            ## Make this look like an R logical does when it is tabulated
+            d[[1]]$name <- "TRUE"
+            d[[2]]$name <- "FALSE"
```
[the remark this answers is not in the document]
Comment (nealrichardson) [reply R/cube-dims.R]: Good point, I'll sort the categories.

[the remark this answers is not in the document]
Comment (nealrichardson) [reply R/cube-dims.R]: Turns out that it's not that simple. We'd have to `aperm` the data too. I don't think it's worth it at this time.

[the remark this answers is not in the document]
Comment (jonkeane) [reply R/cube-dims.R]: Yeah, I agree that mucking with the cube is probably not worth the slight difference in crosstab behavior.

**Pull request merged**